homeassistant/components/notify/hipchat.py

```diff
@@ -0,0 +1,103 @@
+"""
+HipChat platform for notify component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/notify.hipchat/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.notify import (
+    ATTR_TARGET, ATTR_DATA,
+    PLATFORM_SCHEMA, BaseNotificationService)
+from homeassistant.const import (
+    CONF_TOKEN)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['python-simple-hipchat-v2==0.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_COLOR = 'color'
+CONF_ROOM = 'room'
+CONF_NOTIFY = 'notify'
+CONF_FORMAT = 'format'
+CONF_HOST = 'host'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Required(CONF_ROOM): cv.string,
+    vol.Optional(CONF_COLOR, default='yellow'): cv.string,
+    vol.Optional(CONF_NOTIFY, default=False): cv.boolean,
+    vol.Optional(CONF_FORMAT, default='text'): cv.string,
+    vol.Optional(CONF_HOST, default='https://api.hipchat.com/'): cv.string,
+})
+
+
+def get_service(hass, config, discovery_info=None):
+    """Get the HipChat notification service."""
+    return HipchatNotificationService(
+        config[CONF_TOKEN],
+        config[CONF_ROOM],
+        config[CONF_COLOR],
+        config[CONF_NOTIFY],
+        config[CONF_FORMAT],
+        config[CONF_HOST])
+
+
+class HipchatNotificationService(BaseNotificationService):
+    """Implement the notification service for HipChat."""
+
+    _valid_colors = {'yellow', 'green', 'red', 'purple', 'gray', 'random'}
+    _valid_formats = {'text', 'html'}
+
+    def __init__(self, token, default_room, default_color, default_notify,
+                 default_format, host):
+        """Initialize the service."""
+        import hipchat_v2
+        self._token = token
+        self._default_room = default_room
+        if default_color in self._valid_colors:
+            self._default_color = default_color
+        else:
+            self._default_color = 'yellow'
+        if isinstance(default_notify, bool):
```

Comment:
This check isn't necessary. Voluptuous will enforce that only a boolean is provided.